test/services/return-logs/void-no-return-required-licence-return-logs.service.test.js

```diff
@@ -0,0 +1,86 @@
+'use strict'
+
+// Test framework dependencies
+const Lab = require('@hapi/lab')
+const Code = require('@hapi/code')
+
+const { describe, it, before } = (exports.lab = Lab.script())
+const { expect } = Code
+
+// Test helpers
+const LicenceHelper = require('../../support/helpers/licence.helper.js')
+const ReturnVersionHelper = require('../../support/helpers/return-version.helper.js')
+const ReturnLogHelper = require('../../support/helpers/return-log.helper.js')
+
+// Thing under test
+const VoidNoReturnRequiredLicenceReturnLogsService = require('../../../app/services/return-logs/void-no-return-required-licence-return-logs.service.js')
+
+describe('Return Logs - Void No Return Required Licence Return Logs service', () => {
+  let licence
+  let returnVersion
+  let returnLogMatchingVersion
+  let returnLogNotMatchingVersion
+  let returnLogBeingChecked
+
+  describe('when provided a licence version id that has an end date', () => {
+    before(async () => {
+      licence = await LicenceHelper.add()
+      returnVersion = await ReturnVersionHelper.add({
+        licenceId: licence.id,
+        endDate: new Date('2023-03-31'),
+        startDate: new Date('2022-04-01')
+      })
+      returnLogMatchingVersion = await ReturnLogHelper.add({
+        endDate: new Date('2023-03-31'),
+        licenceRef: licence.licenceRef,
+        startDate: new Date('2022-04-01')
+      })
+      returnLogNotMatchingVersion = await ReturnLogHelper.add({
+        endDate: new Date('2024-03-31'),
+        licenceRef: licence.licenceRef,
+        startDate: new Date('2023-04-01')
+      })
+    })
+
+    it('voids the return logs that are between the start and end dates of the no return required version', async () => {
+      await VoidNoReturnRequiredLicenceReturnLogsService.go(
+        licence.licenceRef,
+        returnVersion.startDate,
+        returnVersion.endDate
+      )
+
+      returnLogBeingChecked = await returnLogMatchingVersion.$query()
+      expect(returnLogBeingChecked.status).to.equal('void')
+
+      returnLogBeingChecked = await returnLogNotMatchingVersion.$query()
+      expect(returnLogBeingChecked.status).to.equal('due')
+    })
+  })
```

Comment:
A bit like the file name, I think this description is trying to convey the scenario this would be used in, but is confusing when applied to the service: we don't provide it a licence ID or the licence's end date.

We're also just testing a service that just patches return logs, and doesn't care about the return versions they are linked to. So, I am not sure we need to create any return versions as part of this test. I actually think we don't even need to create a licence record.

So, perhaps ...

- when provided with a licence ref
  - for which there are return logs
    - and no "endDate" is provided
    - and an "endDate" is provided
  - for which there are no return logs

🤷